tests/backup/backup_test.go

```diff
@@ -2661,6 +2661,265 @@ var _ = Describe("{BackupAlternatingBetweenLockedAndUnlockedBuckets}", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			backupName = fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+			timestamp := time.Now().Unix()
+			periodicPolicyName = fmt.Sprintf("%v-%v", "interval", timestamp)
+			log.Infof("Creating backup interval schedule policy - %s", periodicPolicyName)
+			intervalSchedulePolicyInfo := Inst().Backup.CreateIntervalSchedulePolicy(5, 15, 2)
+
+			intervalPolicyStatus := Inst().Backup.BackupSchedulePolicy(periodicPolicyName, uuid.New(), orgID, intervalSchedulePolicyInfo)
+			dash.VerifyFatal(intervalPolicyStatus, nil, fmt.Sprintf("Creating interval schedule policy %v", periodicPolicyName))
+
+			log.Infof("Fetching Schedule uid %v", periodicPolicyName)
+			schPolicyUid, err = Inst().Backup.GetSchedulePolicyUid(orgID, ctx, periodicPolicyName)
+			log.FailOnError(err, "Generating pre rule UID for deployed apps failed for %v", periodicPolicyName)
+
+			//CreateSchedule backup
+			log.InfoD("Backup schedule name - %v", backupName)
+			_, err = CreateScheduleBackup(backupName, SourceClusterName, backupLocationName, backupLocationUID, []string{bkpNamespaces[0]}, nil, orgID, "", "", "", "", periodicPolicyName, schPolicyUid, ctx)
+			log.FailOnError(err, "Creating Schedule Backup")
+		})
+
+		Step("Validate the Access toggle ", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			accesses = append(accesses, ViewOnlyAccess, RestoreAccess, FullAccess)
+
+			// Get user context
+			ctxNonAdmin, err := backup.GetNonAdminCtx(userName, password)
+			log.FailOnError(err, "Fetching %s ctx", userName)
+
+			// Register Source and Destination cluster
+			log.InfoD("Registering Source and Destination clusters from user context")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctxNonAdmin)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+
+			for _, accessLevel := range accesses {
+				log.InfoD("Sharing cluster with %v access to user %s", accessLevel, userName)
+				err := ClusterUpdateBackupShare(SourceClusterName, nil, []string{userName}, accessLevel, true, ctx)
+				log.FailOnError(err, "Failed sharing all backups for cluster [%s]", SourceClusterName)
+
+				userBackups, err := GetAllBackupsForUser(userName, password)
+				log.FailOnError(err, "Fail on Fetching backups for %s", userName)
+
+				restoreName := fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+				ValidateSharedBackupWithUsers(userName, ctxNonAdmin, accessLevel, userBackups[len(userBackups)-1], restoreName)
+				if accessLevel != ViewOnlyAccess {
+					restoreNames = append(restoreNames, restoreName)
+				}
+				log.InfoD("RestoreNames - %v", restoreNames)
+				if accessLevel == FullAccess {
+					log.InfoD("The fullaccess exit begins ")
+					break
+				}
+				//wait for 15 mins
+				log.InfoD("wait for 15 mins")
+				time.Sleep(15 * time.Minute)
```

Comment:
15 minutes seems like a long sleep, what is the purpose of this 15 minute sleep and why exactly 15 minutes, could you please elaborate? Will 15 minutes be always enough? Should we wrap this into a retry instead with some timeout, so we can try more than once and more often? Also don't report this `log.InfoD("wait for 15 mins")` to Aetos dashboard, we don't need to see it there

Reply:
I'm creating a schedule backupup and minimum time there is 15 mins and in this test case we need to toggle the access of shared cluster and check if we can perform operations on the subsequent backups created by schedule after every 15 mins.
Even after 15 mins sleep we are checking if the new backup came up or not and if its in success state or not